core/tests/protractor/editorAndPlayer.js

```diff
@@ -26,11 +26,69 @@ var ExplorationPlayerPage =
 
 describe('Full exploration editor', function() {
   var explorationPlayerPage = null;
+  var parentId2 = null;
+  var refresherExplorationId = null;
+  var currentExplorationId = null;
+  var pathname = null;
+  var parentId1 = null;
 
   beforeAll(function() {
     explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
   });
 
+  it('should redirect back to parent exploration correctly when parent id is ' +
+      'given as query parameter', function() {
+    users.createUser('user1@editorAndPlayer.com', 'user1EditorAndPlayer');
+    users.login('user1@editorAndPlayer.com');
+
+    workflow.createAndPublishExploration(
+      'Parent Exploration 1',
+      'Algebra',
+      'This is the topmost parent exploration.');
+    general.getExplorationIdFromEditor().then(function(explorationId) {
+      parentId1 = explorationId;
+
+      workflow.createAndPublishExploration(
+        'Parent Exploration 1',
+        'Algebra',
+        'This is the second parent exploration to which refresher ' +
+        'exploration redirects.');
+      general.getExplorationIdFromEditor().then(function(explorationId) {
+        parentId2 = explorationId;
+
+        workflow.createAndPublishExploration(
+          'Refresher Exploration',
+          'Algebra',
+          'This is the most basic refresher exploration');
+        general.getExplorationIdFromEditor().then(function(explorationId) {
+          refresherExplorationId = explorationId;
+
+          browser.get('/explore/' + refresherExplorationId + '?parent=' +
+            parentId1 + '&parent=' + parentId2);
+          browser.waitForAngular();
+
+          /* The summary tile for redirection has to be scrolled down in the
```

Comment:
Shouldn't this comment be in clickOnSummaryTileAtEnd()? It lacks context here.

Reply:
Yes, changed it.